### DISCUSSION TITLE
v2 Broadcasting Echo not working, only polling is working for Database Notification. Multi guard issue

### DISCUSSION THREAD

**Discussion by faizananwerali** [filamentphp/filament#7693] [Help]
v2 Broadcasting Echo not working, only polling is working for Database Notification.

Created the table using migration.

set this in the filament.php config file.
```php
'broadcasting' => [
 
    'echo' => [
        'broadcaster' => 'pusher',
        'key' => env('VITE_PUSHER_APP_KEY'),
        'cluster' => env('VITE_PUSHER_APP_CLUSTER'),
        'forceTLS' => true,
    ],
    
],
```

applied keys from pusher.com to the .env file

```dosini
PUSHER_APP_ID=1xxxxxx
PUSHER_APP_KEY=1xxxxxxxxxxxxxxxxxxx
PUSHER_APP_SECRET=9xxxxxxxxxxxxxxxxxxx
PUSHER_HOST=api-us2.pusher.com
PUSHER_PORT=443
PUSHER_SCHEME=https
PUSHER_APP_CLUSTER=us2

VITE_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
VITE_PUSHER_HOST="${PUSHER_HOST}"
VITE_PUSHER_PORT="${PUSHER_PORT}"
VITE_PUSHER_SCHEME="${PUSHER_SCHEME}"
VITE_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"
```

executed the command `php artisan optimize:clear`

generated a notification for admin users.

```php
            $adminUsers = AdminUser::all();
            Notification::make()
                ->title('Inquiry submitted by ' . $contactUs->name)
                ->sendToDatabase($adminUsers)
                ->broadcast($adminUsers);
```

but it is still using long polling even though the pusher is connected via WebSocket.

Tried changing the broadcast driver in the .env file.
```dosini
#BROADCAST_DRIVER=log
BROADCAST_DRIVER=pusher
```

it starts giving a 403 error.
```
POST https://xxxxx.test/broadcasting/auth 403
```

## Admin Model

```php
class AdminUser extends Authenticatable implements FilamentUser, HasAvatar, HasName, MustVerifyEmail
{
    use HasApiTokens, HasFactory, Notifiable, TwoFactorAuthenticatable, HasFilamentShield;
    use QueryCacheable;
    
    // ...

}
```

## Versions:

Laravel v10.16.1
PHP v8.1.19
Filament v2.17.51
Tinker v2.8.1
Livewire v2.12.4

**Comment by danharrin** [comment]
Probably missing the broadcasting route or something for the user

**Comment by faizananwerali** [comment]
That is why I could not figure out what is missing. Can you help me with this?

**Reply by danharrin** [reply]
I'm sorry I cannot help here, it is to do with a misconfiguration in your local project and not Filament

**Reply by faizananwerali** [reply]
It was a multiple guard issue, which was not mentioned in the docs.

**Comment by faizananwerali** [comment]
I even tried this but did not make any difference.

```php
Broadcast::channel('App.Models.User.{id}', function ($user, $id) {
    return (int) $user->id === (int) $id;
});
Broadcast::channel('App.Models.AdminUser.{id}', function ($user, $id) {
    return (int) $user->id === (int) $id;
});
```

**Comment by faizananwerali** [comment]
Finally made it by changing the route middleware.

```php
        Broadcast::routes(['middleware' => ['web','auth:admin_web']]);
```

https://stackoverflow.com/a/50033096/3774282
https://github.com/laravel/framework/issues/16151#issuecomment-257246566
